### PR TITLE
TAS-78 Create, Read, Delete Event From GUI

### PR DESCRIPTION
**Pull request by adrastaea**
# Description
Hooks up the frontend to the backend.

# Impact
* Routers are now used to divide app into modules
* All subpackages properly have an `__init__.py` now
* `/events/` reads db and lets you delete events
* `/events/new/{event_id}` is restyled and can create events
* `/events/{event_id}` displays full event data pulled from db
* changes to schema and model (along with type hints in API path ops)
* invite cards can be deleted
* probably other things

### CODE REVIEW

File: how2meet/ui/frontend.py
Comment (csrubin) [inline how2meet/ui/frontend.py:1]: I feel like this can all just be moved into `main.py` since the frontend is actually being handled by multiple files in `ui/` -- not a big deal, just seems superfluous

File: how2meet/ui/frontend.py
Comment (csrubin) [inline how2meet/ui/frontend.py:30]: I see you changed the `id` fields to `str` but we can probably enforce UUIDs as a type

File: how2meet/ui/pages/events.py
Comment (csrubin) [inline how2meet/ui/pages/events.py:87]: I'm unclear on how this differs from `.bind_value()` 

I prefer what you have here since it is more intuitive, but I'm wary that there's some small but important differences for how they work. Such as when the data updates, whether is reactive after focus has switched or if it requires a "post"

File: how2meet/ui/pages/events.py
Comment (csrubin) [inline how2meet/ui/pages/events.py:81]: Related to previous comment--I'm not sure what we get out of casting the UUID to a string. Pedantic/FastAPI should be capable of data validation on non-standard types.

File: how2meet/ui/pages/events.py
Comment (csrubin) [inline how2meet/ui/pages/events.py:56]: Should use the `BASE_URL` envvar defined elsewhere (main.py?) instead of hardcoding the localhost

File: how2meet/ui/pages/events.py
Comment (csrubin) [inline how2meet/ui/pages/events.py:33]: Not blocking, but from UX perspective I was inclined to try and click on the card itself rather than the small buttons. I think nearly all NiceGUI components support an `.on()` method that can do this

File: how2meet/ui/pages/events.py
Comment (csrubin) [inline how2meet/ui/pages/events.py:114]: ..this smells of something funky. Lots of duplicated effort. Instead, couldn't the `Event` model be imported and used directly here? For example:
```
start_date__input = ui.date()
start_time_input = ui.time()
```
-->
```
event = Event()
event.start_date = ui.date()
event.start_time = ui.time()
```

File: how2meet/ui/pages/events.py
Comment (csrubin) [inline how2meet/ui/pages/events.py:132]: use `BASE_URL`

**Review by csrubin — APPROVED** [review]
Overall so happy with all the progress! Also looks like this encapsulates TAS-79 too!

Largely, I'm inclined not to block merging this early in the project, but you may want to incorporate some of my feedback. Also, I'd wait for @Ben's review as well before merging. 

also also, why do u hate precommit :(

**Comment by csrubin** [comment]
I also noticed you've added loggers (and removed the logging file?) but haven't actually logged anything. Is this intentional? Should there be another ticket for adding logging? I feel like probably not. Same with tests (which are also missing)

File: how2meet/routers/events.py
Comment (csrubin) [inline how2meet/routers/events.py:14]: Return type hints?

File: how2meet/ui/pages/settings.py
Comment (csrubin) [inline how2meet/ui/pages/settings.py:24]: this did not persist across pages for some reason

Comment (adrastaea) [reply how2meet/ui/pages/settings.py:24]: Already been documented in a ticket

File: how2meet/ui/frontend.py
Comment (adrastaea) [reply how2meet/ui/frontend.py:1]: I want to keep a division between the front and back end in case they end up being deployed separately

File: how2meet/ui/pages/events.py
Comment (adrastaea) [reply how2meet/ui/pages/events.py:87]: Consider this unfinished work. This ticket didn't address invites. I just wanted something somewhat working there.

File: how2meet/ui/pages/events.py
Comment (adrastaea) [reply how2meet/ui/pages/events.py:56]: Yes, but I didn't have a clear solution for how that value was passed around so I left it for future work.

File: how2meet/ui/pages/events.py
Comment (adrastaea) [reply how2meet/ui/pages/events.py:33]: Yes, this is just a GUI for testing though. This page probably won't even exist in anything like this form in the final app.

File: how2meet/ui/pages/events.py
Comment (adrastaea) [reply how2meet/ui/pages/events.py:114]: No, because it's so pedantic that if everything isn't perfect then the validation will fail, so you can't do partial instantiations. Validation will be handled by the backend.

File: how2meet/ui/pages/events.py
Comment (csrubin) [reply how2meet/ui/pages/events.py:56]: Fair enough! pls write a ticket :)

File: how2meet/ui/pages/events.py
Comment (csrubin) [reply how2meet/ui/pages/events.py:114]: Ugh that's annoying, I ran into something similar. What if you use the ORM model, not the Pydantic one? This just looks super frustrating to deal with as things change

File: how2meet/ui/pages/events.py
Comment (csrubin) [reply how2meet/ui/pages/events.py:81]: gross

File: how2meet/ui/frontend.py
Comment (csrubin) [reply how2meet/ui/frontend.py:30]: @BenMemberg is unsure about the uuid being generated by the frontend. need a better way to pass it around and save to db at the right time

File: how2meet/main.py
Comment (BenMemberg) [inline how2meet/main.py:30]: I like the /api/ namespacing of the backend 👍

File: how2meet/main.py
Comment (BenMemberg) [inline how2meet/main.py:14]: I read "events" and "invites" as the entities themselves not their routers... could we name them something else or just import each router directly rather than from the parent folder?

File: how2meet/routers/events.py
Comment (BenMemberg) [inline how2meet/routers/events.py:14]: The fact that this file contains both the route definitions _and_ the controller functions they run, I don't know if I like the enclosing folder being "routers". The name feels a little too narrow. Because this kind of _is_ our backend here, not just the router.

File: how2meet/ui/frontend.py
Comment (BenMemberg) [reply how2meet/ui/frontend.py:30]: Yeah I would argue this should be generated by the backend, and the create url would simply be /events/new/ or /events/create/ or something.

File: how2meet/ui/pages/events.py
Comment (BenMemberg) [reply how2meet/ui/pages/events.py:55]: I agree that a file containing the async API functionality is a good idea. I would also agree it doesn't have to be part of this PR. Also also I mentioned this to Connor yesterday but a utils file is probably more appropriate her than a class... I can't think of anything we gain at the moment by instantiating a client class rather than just calling a reusable function.
On the follow-up question... no we probably don't want to expose our internal API for anything for a while.

File: how2meet/ui/pages/events.py
Comment (BenMemberg) [reply how2meet/ui/pages/events.py:113]: I've talked about this with both of you separately I believe, but to summarize: I wouldn't expect us to have to fit input data to a schema or type outside of individual field validators in the frontend. I would expect the client to simply send a JSON of data to the server, which is then responsible for validating the data and then calling a model create to create a DB entry.

**Review by BenMemberg — APPROVED** [review]
Great start! I'll probably branch off of this and start gunking around with some stuff to figure out these frameworks for myself 😈

File: how2meet/routers/events.py
Comment (adrastaea) [reply how2meet/routers/events.py:14]: Maybe we just call the enclosing folder `api`?

**Pull request merged**
